libecole/tests/src/reward/test-nnodes.cpp

```diff
@@ -0,0 +1,85 @@
+#include <memory>
+#include <string>
+#include <tuple>
+
+#include <catch2/catch.hpp>
+
+#include "ecole/environment/branching.hpp"
+#include "ecole/observation/nothing.hpp"
+#include "ecole/reward/nnodes.hpp"
+
+#include "conftest.hpp"
+#include "reward/unit-tests.hpp"
+
+using namespace ecole;
+
+TEST_CASE("NNodes unit tests", "[unit][reward]") {
+	reward::unit_tests(reward::NNodes{});
+}
+
+TEST_CASE(
+	"NNodes returns the difference in the total number of processed nodes between two states",
+	"[reward]") {
+
+	auto reward_func = reward::NNodes{};
+	auto model = get_solving_model();
+	reward_func.reset(model);
+
+	SECTION("NNodes is positive") { REQUIRE(reward_func.obtain_reward(model) >= 0); }
+
+	SECTION("NNodes is zero if nothing happended between two states") {
+		reward_func.obtain_reward(model);
+		REQUIRE(reward_func.obtain_reward(model) == 0);
```

Comment:
Don't we need two calls to `obtain_reward` to ensure that effectively nothing happened between two states?

Reply:
That's right, I though the previous SECTION code would have been executed before.